internal/datacoord/index_meta.go

```diff
@@ -969,3 +969,10 @@ func (m *indexMeta) AreAllDiskIndex(collectionID int64, schema *schemapb.Collect
 	allDiskIndex := len(vectorFields) == len(vectorFieldsWithDiskIndex)
 	return allDiskIndex
 }
+
+func (m *indexMeta) HasIndex(collectionID int64) bool {
```

Comment:
Be more precise; it is necessary to determine here whether the index information exists, as it may have already been designated as IsDeleted.